uberenv.py

```diff
@@ -854,20 +854,33 @@ def clone_repo(self):
             sexe("git stash", echo=True)
             res = sexe("git pull", echo=True)
             if res != 0:
-                #Usually untracked files that would be overwritten
+                # Usually untracked files that would be overwritten
                 print("[ERROR: Git failed to pull]")
                 sys.exit(-1)
 
-        # Move destination of Spack builtin package repository if not included in Spack repo (pre-1.0)
+        # Move and checkout Spack builtin package repository if not included in Spack repo
         if not os.path.exists(pjoin(self.dest_spack, "var", "spack", "repos", "builtin")):
-            builtin_repo = pjoin(self.dest_dir, "builtin_spack_packages_repo")
-            print(f"[info: moving spack builtin package repository to {builtin_repo}]")
-            spack_repo_set_cmd = f"{self.spack_exe(use_spack_env=False)} repo set --destination {builtin_repo} builtin"
+            packages_repo = pjoin(self.dest_dir, "builtin_spack_packages_repo")
+
+            print(f"[info: moving spack builtin package repository to {packages_repo}]")
+            spack_repo_set_cmd = f"{self.spack_exe(use_spack_env=False)} repo set --destination {packages_repo} builtin"
             res = sexe(spack_repo_set_cmd, echo=True)
             if res != 0:
                 print("[ERROR: Failed to set builtin package repository destination]")
                 sys.exit(-1)
 
+            # Optionally, check out Spack's builtin package repo to a specific commit
+            if "spack_packages_commit" in self.project_args:
+                sha1 = self.project_args["spack_packages_commit"]
```

Comment:
I don't remember how this works.
Does this take values from our `.uberenv_config.json` file, or do we need explicitly add/handle entries there?

If not, please add support for having `spack_packages_commit` in `.uberenv_config.json`

Reply:
It takes it from `.uberenv_config.json`.